campaignion_wizard/src/WebformStep.php

```diff
@@ -33,8 +33,42 @@ public function pageCallback() {
 
     // Build form for webform_template select box.
     if (module_exists('campaignion_action_template')) {
+      // Check if there are existing form submissions.
+      $submissions_exist = FALSE;
+      $sql = <<<SQL
+SELECT count(*) AS submissions
+FROM {webform_submissions}
+WHERE nid=:nid
+SQL;
+      $submissions = db_query($sql, array(':nid' => $this->wizard->node->nid));
+      foreach ($submissions as $s) {
+        if ($s->submissions > 0) {
+          $submissions_exist = TRUE;
+        }
+      }
+
       $build[] = drupal_get_form('campaignion_action_template_selector_form', $this->wizard->node)
```

Comment:
I think it would be better to generate the form into a temporary variable first, then modify it, and then assign it to `$build[]`. This way we would avoid to assign values to a "magic" index.